x-pack/plugins/ml/server/saved_objects/checks.ts

```diff
@@ -163,201 +161,6 @@ export function checksFactory(
-        } else {
-          const df = datafeeds.datafeeds.find((d) => d.job_id === job.jobId);
-          const jobId = job.jobId;
-          const datafeedId = df?.datafeed_id;
-
-          tasks.push(async () => {
-            try {
-              if (datafeedId !== undefined) {
-                await jobSavedObjectService.addDatafeed(datafeedId, jobId);
-              }
-              results.datafeedsAdded[job.jobId] = { success: true };
-            } catch (error) {
-              results.datafeedsAdded[job.jobId] = { success: false, error };
-            }
-          });
-        }
-      } else if (
-        job.checks.jobExists === true &&
-        job.checks.datafeedExists === false &&
-        job.datafeedId !== null &&
-        job.datafeedId !== undefined
-      ) {
-        // remove datafeed id for jobs where the datafeed no longer exists but the id is populated in the saved object
-        if (simulate === true) {
-          results.datafeedsRemoved[job.jobId] = { success: true };
-        } else {
-          const datafeedId = job.datafeedId;
-          tasks.push(async () => {
-            try {
-              await jobSavedObjectService.deleteDatafeed(datafeedId);
-              results.datafeedsRemoved[job.jobId] = { success: true };
-            } catch (error) {
-              results.datafeedsRemoved[job.jobId] = { success: false, error: error.body ?? error };
-            }
-          });
-        }
-      }
-    }
-    await Promise.allSettled(tasks.map((t) => t()));
-    return results;
-  }
-
-  async function initSavedObjects(simulate: boolean = false, namespaces: string[] = ['*']) {
-    const results: { jobs: Array<{ id: string; type: string }>; success: boolean; error?: any } = {
-      jobs: [],
-      success: true,
-    };
-    const status = await checkStatus();
-
-    const jobs: JobObject[] = [];
-    const types: JobType[] = ['anomaly-detector', 'data-frame-analytics'];
-
-    types.forEach((type) => {
-      status.jobs[type].forEach((job) => {
-        if (job.checks.savedObjectExits === false) {
-          if (simulate === true) {
-            results.jobs.push({ id: job.jobId, type });
-          } else {
-            jobs.push({
-              job_id: job.jobId,
-              datafeed_id: job.datafeedId ?? null,
-              type,
-            });
-          }
-        }
-      });
-    });
-    try {
-      const createResults = await jobSavedObjectService.bulkCreateJobs(jobs, namespaces);
-      createResults.saved_objects.forEach(({ attributes }) => {
-        results.jobs.push({
-          id: attributes.job_id,
-          type: attributes.type,
-        });
-      });
-    } catch (error) {
-      results.success = false;
-      results.error = Boom.boomify(error).output;
-    }
-    return results;
-  }
-
   async function _loadAllJobSavedObjects() {
     const { body } = await client.asInternalUser.search<SearchResponse<SavedObjectJob>>({
```

Comment:
I used a search here because I need a list of all saved objects across all spaces, even the ones the user doesn't have access to. As far as i'm aware the saved object client can't do that?
`checkStatus` needs to know if a job has a saved object, otherwise we'll end up with duplicates when repairing.

Reply:
@legrego i've updated these searches so it now uses the saved objects client